### PR TITLE
Fixed crash when trying to drop ENR relations during subtransaction.

### PR DESCRIPTION
**Pull request by lejaokri**
Unable to drop ENR tables in subtransaction because invalidation messages was never accepted failing the check in
PrepareInvalidationState().

This is because ENR relations skip GetCurrentCommandId(true) hence making CommandCounterIncrement() practically a no-op.

Also fix a Day 1 issue when caller of pltsql_clean_table_variables() skips some important error handling when pltsql_clean_table_variables fails.

Task: BABEL-4737

### Check List
- [ ] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is under the terms of the Apache 2.0 and PostgreSQL licenses, and grant any person obtaining a copy of the contribution permission to relicense all or a portion of my contribution to the PostgreSQL License solely to contribute all or a portion of my contribution to the PostgreSQL open source project.

For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/babelfish-for-postgresql/babelfish_extensions/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: test/JDBC/expected/table_variable_xact_errors_isolation_snapshot.out [hunk 1]
```diff
@@ -583,10 +583,6 @@ AS
     DELETE trgt FROM Discount trgt           -- Discount does not exist
     COMMIT
 go
-~~ERROR (Code: 2714)~~
```
Comment (surendravishnoi) [inline test/JDBC/expected/table_variable_xact_errors_isolation_snapshot.out:586]: why this change?

**Review by surendravishnoi — APPROVED** [review]

File: test/JDBC/expected/table_variable_xact_errors_isolation_snapshot.out (hunk 1 above)
Comment (lejaokri) [reply test/JDBC/expected/table_variable_xact_errors_isolation_snapshot.out:586]: I am adding a drop of the function so the error does not exist anymore.

```
Message: Function 'usp_populatediscount' already exists with the same name
```
